src/player.py

```diff
@@ -1,43 +1,77 @@
 from __future__ import annotations
 
 import random
+from typing import Sequence
 
 from arena.spawn_point import SpawnPoint
 from player_input import PlayerInput
+from rounds.game_modes.game_mode import GameModePlayerState
+from rounds.round_controller import RoundController
 from sprite_lists import SpriteLists
 from vehicle import Vehicle
 
 
 class Player:
+
+    round_controller: RoundController
+    "Necessary for player to notify the round controller and game mode about relevant events"
+
```

Comment:
Hmm, yeah that's confusing.  It doesn't need access to `round_controller`, but it needs `game_mode`.  And `vehicle` is the one using it, since it's the thing that makes death happen, and it needs to notify the gamemode.

https://github.com/CastIronChat/twisted-metal/blob/59abae8603953be922d71c2563b9cd8267c76a36/src/vehicle.py#L96

I tweaked it to store `game_mode` instead of `round_controller`, to be a bit clearer.

Reply:
Ah, that's were it's called. Yeah, storing game_mode seems cleaner.